docs/content/contributing/tests.md

```diff
@@ -0,0 +1,135 @@
+
+    The default value of `BATS_PARALLEL_JOBS` is 2. Increasing it to `3` requires 6 threads and 6GB of main memory; increasing it to `4` requires 8 threads and at least 8GB of main memory.
+
+!!! warning "Test Output when Running in Parallel"
+
+    [When running tests in parallel][docs-bats-parallel] (_with `make clean generate-accounts tests/parallel/setX`_), BATS will delay outputting the results until completing all test cases within a file.
+
+    This also delays test failures as a result. When troubleshooting parallel set tests, you may prefer to run them serially as advised below.
+
+    When writing tests, ensure that parallel set tests still pass when run in parallel. You need to account for other tests running in parallel that may interfere with your own tests logic.
+
+### An Example
+
+Let's say you adjusted Rspamd and want to see whether you did not introduce a regression. Of course, you already made up your mind and did your best to ensure that beforehand. You want to run only the Rspamd test in the beginning, so you use:
+
+```console
+$ make clean generate-accounts test/rspamd
+rspamd.bats
+ ✓ [Rspamd] Postfix's main.cf was adjusted [12]
+ ✓ [Rspamd] normal mail passes fine [44]
+ ✓ [Rspamd] detects and rejects spam [122]
+ ✓ [Rspamd] detects and rejects virus [189]
+```
+
+But then you realize that the Rspamd test is somehow also testing ClamAV functionality. You also want to test ClamAV, and because you made changes, you want to run the Rspamd test again, so you go ahead:
+
+```console
+$ make clean generate-accounts test/rspamd,clamav
+rspamd.bats
+ ✓ [Rspamd] Postfix's main.cf was adjusted [12]
+ ✓ [Rspamd] normal mail passes fine [44]
+ ✓ [Rspamd] detects and rejects spam [122]
+ ✓ [Rspamd] detects and rejects virus [189]
+clamav.bats
+ ✓ [ClamAV] log files exist at /var/log/mail directory [68]
+ ✓ [ClamAV] should be identified by Amavis [67]
+ ✓ [ClamAV] freshclam cron is enabled [76]
+ ✓ [ClamAV] env CLAMAV_MESSAGE_SIZE_LIMIT is set correctly [63]
+ ✓ [ClamAV] rejects virus [60]
+```
+
+You're mostly finished, so want to check the parallel set that Rspamd and ClamAV tests belong to:
+
+```console
+$ make clean generate-accounts tests/parallel/set1
+default_relay_host.bats
+ ✓ [Relay] (ENV) 'DEFAULT_RELAY_HOST' should configure 'main.cf:relayhost' [88]
+spam_virus/amavis.bats
+ ✓ [Amavis] SpamAssassin integration should be active [1165]
+spam_virus/clamav.bats
+ ✓ [ClamAV] log files exist at /var/log/mail directory [73]
+ ✓ [ClamAV] should be identified by Amavis [67]
+ ✓ [ClamAV] freshclam cron is enabled [76]
+...
+```
+
+In the end, before opening the PR, you want to test everything because you want to give maintains an easy time reviewing your nice PR, so you run:
+
+```console
+$ make clean tests
+...
+```
+
+[//]: # (Links)
```

Comment:
? Is this doing anything in particular?

Reply:
Just a markdown comment; I can remove it though.